how-to/integrate-with-sso/integrate-with-okta/client/src/auth.ts

```diff
@@ -215,56 +305,152 @@ async function showWindow(url: string): Promise<OpenFin.Window> {
 	});
 }
 
-function saveProperty(propName: string, value: string): void {
-	window.localStorage.setItem(`${STORAGE_REALM}/${propName}`, value);
+/**
+ * Expire the access token by removing it from storage.
+ * The verification polling will then logout.
+ */
+export async function expireAccessToken(): Promise<void> {
+	const accessToken = loadProperty(STORE_ACCESS_TOKEN);
+	if (accessToken) {
+		try {
+			await fetch(`https://${oktaSettings?.domain}/oauth2/v1/revoke`, {
+				method: "POST",
+				headers: {
+					Accept: "application/json",
+					"Content-Type": "application/x-www-form-urlencoded"
+				},
+				body: `client_id=${oktaSettings?.clientId}&token_type_hint=access_token&token=${accessToken}`
+			});
+		} catch (err) {
+			informationCallback("Revoke failed");
+			informationCallback(handleError(err));
+		} finally {
+			removeProperty(STORE_ACCESS_TOKEN);
+		}
+	}
 }
 
-function removeProperty(propName: string): void {
-	window.localStorage.removeItem(`${STORAGE_REALM}/${propName}`);
+/**
+ * The authentication state has changed to restart polling if necessary.
+ * @param isAuthenticated Is the user authenticated.
+ * @param userDetails Details for the logged in user.
+ */
+async function authenticatedStateChanged(
+	isAuthenticated: boolean,
+	userDetails?: { [id: string]: unknown }
+): Promise<void> {
+	pollTimerStop();
+	await authenticatedCallback(isAuthenticated, userDetails);
+	pollTimerStart(isAuthenticated);
 }
 
-async function checkForUrls(win: OpenFin.Window, urls: string[]): Promise<URL | undefined> {
+/**
+ * Stop the token verification polling timer.
+ */
+function pollTimerStop(): void {
+	if (pollTimerId) {
+		window.clearInterval(pollTimerId);
+		pollTimerId = undefined;
+	}
+}
+
+/**
+ * Start the token verification polling timer.
+ * @param isAuthenticated Is the user authenticated.
+ */
+function pollTimerStart(isAuthenticated: boolean): void {
+	if ((oktaSettings?.verifyPollMs ?? 0) > 0 && isAuthenticated) {
+		pollTimerId = window.setInterval(async () => {
+			const accessToken = loadProperty(STORE_ACCESS_TOKEN);
+			if (accessToken) {
+				const userInfo = await checkTokenValidity(accessToken, false);
+				if (!userInfo) {
+					informationCallback("Access token no longer valid, logout");
+					removeProperty(STORE_ACCESS_TOKEN);
+					await authenticatedStateChanged(false);
+				} else {
+					informationCallback("Access token still valid");
+				}
+			} else {
+				informationCallback("Access token not found, logout");
+				await authenticatedStateChanged(false);
+			}
+		}, oktaSettings?.verifyPollMs);
+	}
+}
+
+/**
+ * Check to see if the window is displaying one of the specified urls with a partial match.
+ * @param win The window to check.
+ * @param urls The urls to look for.
+ * @returns The complete url if it has one it was looking for.
+ */
+async function checkForUrls(
+	win: OpenFin.Window | undefined,
+	urls: (string | undefined)[]
+): Promise<URL | undefined> {
```

Comment:
Can we use the new web navigation event instead of polling introduced in https://developer.openfin.co/versions/?product=Runtime#/?product=Runtime&version=30.110.74.8

Reply:
Done